### ISSUE TITLE
predict error: Length of passed values is..., index implies 26

### ISSUE THREAD

**Issue by LakshmikanthanSrikanthan** [facebook/prophet#808]
**Any help on the following. This didn't happen before the previous update of fbprophet.**

> 
> 
> Traceback (most recent call last):
>   File "forecast.py", line 19, in <module>
>     forecast = m.predict(future)
>   File "lib/python3.6/site-packages/fbprophet/forecaster.py", line 1138, in predict
>     intervals = self.predict_uncertainty(df)
>   File "lib/python3.6/site-packages/fbprophet/forecaster.py", line 1340, in predict_uncertainty
>     sim_values = self.sample_posterior_predictive(df)
>   File "lib/python3.6/site-packages/fbprophet/forecaster.py", line 1304, in sample_posterior_predictive
>     s_m=component_cols['multiplicative_terms'],
>   File "lib/python3.6/site-packages/fbprophet/forecaster.py", line 1372, in sample_model
>     Xb_a = np.matmul(seasonal_features.values, beta * s_a) * self.y_scale
>   File "lib/python3.6/site-packages/pandas/core/series.py", line 648, in __array_wrap__
>     copy=False).__finalize__(self)
>   File "lib/python3.6/site-packages/pandas/core/series.py", line 262, in __init__
>     .format(val=len(data), ind=len(index)))
> ValueError: Length of passed values is 3106, index implies 26
> 
> Process finished with exit code 1

**Comment by lppier** [comment]
Just a wild guess, could it be that the index was not reset? 
https://www.geeksforgeeks.org/python-pandas-dataframe-reset_index/

**Comment by yohannj** [comment]
TL;DR Force the use of the following lib version:
pyparsing==2.3.0
tqdm==4.29.0
numpy==1.15.4

I faced the same issue.
I rebuild a code which tests were working 3 days ago, and today those same test with the same code were failing.
Comparing the build logs, I found that a new version was used for 3 dependencies (pyparsing 2.3.0->2.3.1, tqdm 4.29.0->4.29.1, numpy 1.15.4->1.16.0). I didn't investigated more than that, just forced their version to the one that was automatically downloaded last Friday, and it solved the issue.

**Comment by Frans06** [comment]
Last Numpy version not supported yet. Use 1.15.4.

**Comment by yohannj** [comment]
Should we try to provide more stability, limiting such issues from happening?

**Comment by aoliong** [comment]
I met the same problem. But when I use numpy==1.15.4 and fbprophet==0.4.post1,  I met another problem:

`---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-068fc0cb780e> in <module>
      1 m = Prophet(daily_seasonality=True)
----> 2 m.fit(df)

~/miniconda3/envs/env01/lib/python3.7/site-packages/fbprophet/forecaster.py in fit(self, df, **kwargs)
   1095             args.update(kwargs)
   1096             try:
-> 1097                 params = model.optimizing(**args)
   1098             except RuntimeError:
   1099                 if 'algorithm' not in args:

~/miniconda3/envs/env01/lib/python3.7/site-packages/pystan/model.py in optimizing(self, data, seed, init, sample_file, algorithm, verbose, as_vector, **kwargs)
    509             data = {}
    510         seed = pystan.misc._check_seed(seed)
--> 511         fit = self.fit_class(data, seed)
    512 
    513         m_pars = fit._get_param_names()

AttributeError: 'StanModel' object has no attribute 'fit_class'`

I follow this [Quick start](https://facebook.github.io/prophet/docs/quick_start.html#python-api),  and other lib version is pystan==2.18.1.0

**Comment by aoliong** [comment]
SOLVED

I found #599 . After reinstall pystan and fbphrophet, it's all right.

**Comment by LakshmikanthanSrikanthan** [comment]
@yohannj @Frans06 @aoliong  Thank you Very Much. The error that @aoliong  pointed out occurred to me too. Trying the fix of downgrading pystan to 2.18.1.0 and also to 2.17.1.0 didn't work for me.  

> > SOLVED
> >
> >I found #599 . After reinstall pystan and fbphrophet, it's all right.

Any other leads ? You can check the error below.

> venv/bin/python /home/matrix/PycharmProjects/Stock_Linux/Syntax/StockAnalyzer/forecast.py
> WARNING:pystan:numpy.ufunc size changed, may indicate binary incompatibility. Expected 216 from C header, got 192 from PyObject
> WARNING:pystan:Something went wrong while unpickling the StanModel. Consider recompiling.
>           ds       y
> 0 2007-11-05  5861.0
> 1 2007-11-06  5822.1
> 2 2007-11-07  5762.0
> 3 2007-11-08  5720.0
> 4 2007-11-12  5642.5
> INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
> Traceback (most recent call last):
>   File "StockAnalyzer/forecast.py", line 15, in <module>
>     m.fit(stock)
>   File "venv/lib/python3.6/site-packages/fbprophet/forecaster.py", line 1097, in fit
>     params = model.optimizing(**args)
>   File "venv/lib/python3.6/site-packages/pystan/model.py", line 511, in optimizing
>     fit = self.fit_class(data, seed)
> AttributeError: 'StanModel' object has no attribute 'fit_class'
> 
> Process finished with exit code 1

**Comment by Leabre** [comment]
Hey ! I met the same issue. Downgrading numpy to 1.14.5 and fbprophet to 0.3.post2 works for me

**Comment by LakshmikanthanSrikanthan** [comment]
> Hey ! I met the same issue. Downgrading numpy to 1.14.5 and fbprophet to 0.3.post2 works for me

Thanks that Worked.

**Comment by LakshmikanthanSrikanthan** [comment]
**_Thanks Everyone for the feedback and help in solving the errors. I am thinking of keeping this thread Open so that we could look up to the dependencies of the packages and their functionality being bugged when upgrading to the latest version of prophet , stan and so on. If you guys suggest otherwise I would close the thread._**

**Comment by bletham** [comment]
I was able to produce this error solely by upgrading numpy from 1.14.5 to 1.16.0

**Comment by bletham** [comment]
The issue is in the numpy/pandas interface. This code works in numpy 1.15.4 but produces this error in 1.16.0:
```
import numpy as np
import pandas as pd
X = np.random.rand(5, 2)
beta = np.random.rand(2)
s_a = pd.Series(np.ones(2))
np.matmul(X, beta * s_a)
```

I'll push a workaround in fbprophet.

**Comment by bletham** [comment]
The fix has been pushed to pypi. Get the latest version (0.4.post2) with 
```
pip install --upgrade fbprophet
```
and it is compatible with both old and new versions of numpy.

**Comment by LakshmikanthanSrikanthan** [comment]
@bletham Thanks for the bug fix. Have a Great Day.

**Comment by bletham** [comment]
I'll leave this open for a little bit since other people may run into this as they upgrade numpy, and hopefully will be able to quickly find this as the solution.

**Comment by wagner-rodeski** [comment]
I'm using fbprophet on kaggle, i did it today (jan 21), and i'm facing the problem related to this thread subject.
Could anyone help me solving it on kaggle?

**Comment by Rym96** [comment]
I am having the same issue 
I tried everything you said and I keep on having this error 

```
 ---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-28-6a35f92430e7> in <module>()
      1 futur = data_prophet.make_future_dataframe(periods=365 * 2, freq='D')
----> 2 dvlptstaff_forecast = data_prophet.predict(futur)

/opt/conda/lib/python3.6/site-packages/fbprophet/forecaster.py in predict(self, df)
   1040             't_change': self.changepoints_t,
   1041             'X': seasonal_features,
-> 1042             'sigmas': prior_scales,
   1043             'tau': self.changepoint_prior_scale,
   1044             'trend_indicator': int(self.growth == 'logistic'),

/opt/conda/lib/python3.6/site-packages/fbprophet/forecaster.py in predict_uncertainty(self, df)
   1242 
   1243         Parameters
-> 1244         ----------
   1245         df: Prediction dataframe.
   1246 

/opt/conda/lib/python3.6/site-packages/fbprophet/forecaster.py in sample_posterior_predictive(self, df)
   1206         # Get cumulative rate and offset at each t
   1207         k_t = k * np.ones_like(t)
-> 1208         m_t = m * np.ones_like(t)
   1209         for s, t_s in enumerate(changepoint_ts):
   1210             indx = t >= t_s

/opt/conda/lib/python3.6/site-packages/fbprophet/forecaster.py in sample_model(self, df, seasonal_features, iteration, s_a, s_m)
   1274     def sample_posterior_predictive(self, df):
   1275         """Prophet posterior predictive samples.
-> 1276 
   1277         Parameters
   1278         ----------

/opt/conda/lib/python3.6/site-packages/pandas/core/series.py in __array_wrap__(self, result, context)
    646         """
    647         return self._constructor(result, index=self.index,
--> 648                                  copy=False).__finalize__(self)
    649 
    650     def __array_prepare__(self, result, context=None):

/opt/conda/lib/python3.6/site-packages/pandas/core/series.py in __init__(self, data, index, dtype, name, copy, fastpath)
    260                             'Length of passed values is {val}, '
    261                             'index implies {ind}'
--> 262                             .format(val=len(data), ind=len(index)))
    263                 except TypeError:
    264                     pass

ValueError: Length of passed values is 737, index implies 20

```

**Comment by bletham** [comment]
@Rym96 can you check that you have version 0.4.post2 installed? In particular, what does
```
pip show fbprophet
```
return?

**Comment by Rym96** [comment]
@bletham  This is what it shows
`Name: fbprophet
Version: 0.3.post2
Summary: Automatic Forecasting Procedure
Home-page: https://facebook.github.io/prophet/
Author: Sean J. Taylor <sjt@fb.com>, Ben Letham <bletham@fb.com>
Author-email: sjt@fb.com
License: BSD
Location: /opt/conda/lib/python3.6/site-packages
Requires: matplotlib, pystan, numpy, pandas, Cython
Required-by: `

**Comment by bletham** [comment]
Ah, I didn't notice before that you're using conda. In that case you need to make sure that the package you are using in conda is 0.4.post2, which can be different from what system python and pip are using.

Can you in conda do
```
import fbprophet
print(fbprophet.__version__)
```
and then try to re-install with conda-forge (which does have 0.4.post2):
```
conda install -c conda-forge fbprophet
```

**Comment by Rym96** [comment]
I can on use it if my period is equal to 13

**Comment by bletham** [comment]
I'm not certain that the version that is actually being loded is 0.4.post2. Could you paste the output of
```
conda list
```
and could you also paste the output of
```
import inspect
from fbprophet import Prophet
print(inspect.getsource(Prophet.sample_model))
```

**Comment by anu0012** [comment]
@bletham In my case output of conda list is 0.4.post2 but I am still facing this issue.

**Comment by bletham** [comment]
@anu0012 can you paste the full traceback for the error, like in the post at the top of this issue?

**Comment by anu0012** [comment]
Hey, @bletham Thanks for the reply. But the issue is now solved. I updated it to the latest version.

**Comment by vsagar8888** [comment]
I am getting below error while executing the line "forecast = m.predict(future)"
versions used:
>>> import numpy
>>> numpy.__version__
'1.16.0'
>>> import fbprophet
fbprophet.>>> fbprophet.__version__
'0.4'


---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-47-e14a83013b80> in <module>()
----> 1 forecast = m.predict(future)

/sys_apps_01/python/python342/lib/python3.4/site-packages/fbprophet/forecaster.py in predict(self, df)
   1144         df['trend'] = self.predict_trend(df)
   1145         seasonal_components = self.predict_seasonal_components(df)
-> 1146         intervals = self.predict_uncertainty(df)
   1147 
   1148         # Drop columns except ds, cap, floor, and trend

/sys_apps_01/python/python342/lib/python3.4/site-packages/fbprophet/forecaster.py in predict_uncertainty(self, df)
   1346         Dataframe with uncertainty intervals.
   1347         """
-> 1348         sim_values = self.sample_posterior_predictive(df)
   1349 
   1350         lower_p = 100 * (1.0 - self.interval_width) / 2

/sys_apps_01/python/python342/lib/python3.4/site-packages/fbprophet/forecaster.py in sample_posterior_predictive(self, df)
   1310                     iteration=i,
   1311                     s_a=component_cols['additive_terms'],
-> 1312                     s_m=component_cols['multiplicative_terms'],
   1313                 )
   1314                 for key in sim_values:

/sys_apps_01/python/python342/lib/python3.4/site-packages/fbprophet/forecaster.py in sample_model(self, df, seasonal_features, iteration, s_a, s_m)
   1378 
   1379         beta = self.params['beta'][iteration]
-> 1380         Xb_a = np.matmul(seasonal_features.values, beta * s_a) * self.y_scale
   1381         Xb_m = np.matmul(seasonal_features.values, beta * s_m)
   1382 

/sys_apps_01/python/python342/lib/python3.4/site-packages/pandas/core/series.py in __array_wrap__(self, result, context)
    502         """
    503         return self._constructor(result, index=self.index,
--> 504                                  copy=False).__finalize__(self)
    505 
    506     def __array_prepare__(self, result, context=None):

/sys_apps_01/python/python342/lib/python3.4/site-packages/pandas/core/series.py in __init__(self, data, index, dtype, name, copy, fastpath)
    264                                        raise_cast_failure=True)
    265 
--> 266                 data = SingleBlockManager(data, index, fastpath=True)
    267 
    268         generic.NDFrame.__init__(self, data, fastpath=True)

/sys_apps_01/python/python342/lib/python3.4/site-packages/pandas/core/internals.py in __init__(self, block, axis, do_integrity_check, fastpath)
   4395         if not isinstance(block, Block):
   4396             block = make_block(block, placement=slice(0, len(axis)), ndim=1,
-> 4397                                fastpath=True)
   4398 
   4399         self.blocks = [block]

/sys_apps_01/python/python342/lib/python3.4/site-packages/pandas/core/internals.py in make_block(values, placement, klass, ndim, dtype, fastpath)
   2950                      placement=placement, dtype=dtype)
   2951 
-> 2952     return klass(values, ndim=ndim, fastpath=fastpath, placement=placement)
   2953 
   2954 # TODO: flexible with index=None and/or items=None

/sys_apps_01/python/python342/lib/python3.4/site-packages/pandas/core/internals.py in __init__(self, values, placement, ndim, fastpath)
    118             raise ValueError('Wrong number of items passed %d, placement '
    119                              'implies %d' % (len(self.values),
--> 120                                              len(self.mgr_locs)))
    121 
    122     @property

ValueError: Wrong number of items passed 3270, placement implies 26.



Will this issue is solved by upgrading fbprophet or it is different issue ?

**Comment by Sonakshi2018** [comment]
@bletham Thanks for the reply. I updated fbprophet to 0.5 and it worked.
